src/PresentationalComponents/Common/Tables.js

```diff
@@ -49,9 +50,19 @@ export const buildTagFilter = (tagFilters) => {
   };
 };
 
-const mapUpdateMethodFilterToAPISpec = (filters) => {
-  //If none option is chosen
-  if (filters?.update_method?.includes('none')) {
+export const mapUpdateMethodFilterToAPISpec = (filters) => {
+  if (
+    filters?.update_method === '' ||
+    isEqual(filters?.update_method?.split(',').sort(), [
+      'dnfyum',
+      'none',
+      'ostree',
+    ])
+  ) {
+    //when user deselects all update_method filters remove the both update_method and impacting filters
+    delete filters.update_method;
```

Comment:
I think we should prefer returning a new object rather than mutating `filters`. We have seen these `delete`s cause issues and unexpected states/results.

Reply:
Yup, actually there is a new object created to return [here](https://github.com/RedHatInsights/insights-advisor-frontend/blob/2e2897c9889e78a047ed9284f89e1029273cfb6a/src/PresentationalComponents/Common/Tables.js#L89). Should I still create a new filter return here as well? Happy to do the best approach, let me know

Reply:
I think Sebastian is correct. You already importing lodash in this file, so you can use something like deepClone method to avoid changing the initial filters object
[_.deepClone](https://www.geeksforgeeks.org/lodash-_-clonedeep-method/)

Reply:
@Fewwy I have made the change, please take a look!

Reply:
@mkholjuraev @Fewwy It would have also been good to completely avoid the `delete` and not need to clone the object. Using spreading and destructing the same can be achieved and it creates a new object, without cloning. See for example: https://javascript.plainenglish.io/avoid-the-delete-keyword-in-javascript-87ff2a47f26c

Reply:
I will keep this in mind and will do the required change as soon as I have  a chance